### PR TITLE
chore(explore): make datasource search input sticky

### LINKED ISSUE #12345: [Explore]make data panel search box sticky to top on scroll
Committed enhancement: setting the search box sticky to top on scroll

Although both sections(Metrics and Columns) of the Data panel display a maximum of 50 entries, they can easily exceed the height of a page. Once entries exceed page height, users need to scroll up and down the page to find the search box, which causes some inconvenience. 

ezgif-6-87b22df12ca3

### PR DESCRIPTION
**Pull request by pkdotson**
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/17326228/104073891-b3af1300-51c3-11eb-8ac8-8b2834bd6e12.mov

Closes #12345 
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/components/DatasourcePanel.tsx [hunk 1]
```diff
@@ -182,13 +183,13 @@ const DataSourcePanel = ({
         actions={actions}
         formData={datasourceControl.mapStateToProps}
       />
+      <input
+        type="text"
+        onChange={search}
+        className="form-control input-sm"
+        placeholder={t('Search Metrics & Columns')}
+      />
```
Comment (ktmud) [inline superset-frontend/src/explore/components/DatasourcePanel.tsx:191]: Can we use AntD input here？The style look too bootstrap-ish.

**Review by graceguo-supercat — APPROVED** [review]
LGTM

**Review by ktmud — APPROVED** [review]

File: superset-frontend/src/explore/components/DatasourcePanel.tsx [hunk 2]
```diff
@@ -131,7 +131,8 @@ const DatasourceContainer = styled.div`
     color: ${({ theme }) => theme.colors.grayscale.light1};
   }
   .form-control.input-sm {
-    margin-bottom: 0;
+    width: 96%;
```
Comment (rusackas) [inline superset-frontend/src/explore/components/DatasourcePanel.tsx:134]: This 96% jumped out at me as a little funny... is there supposed to be some number of pixels on each side of the element? Maybe something like `width: calc(100% - ${({ theme }) => theme.gridUnit * 2}px);` would make sense (or whatever the gutter measurement should be?)

**Review by junlincc — APPROVED** [review]
LGTM, met the request in #12345

File: superset-frontend/src/explore/components/DatasourcePanel.tsx (hunk 1 above)
Comment (rusackas) [reply superset-frontend/src/explore/components/DatasourcePanel.tsx:191]: Let's follow up on that in a separate PR. We should convert ALL inputs to AntD inputs, and just make a big sweep of 'em.

File: superset-frontend/src/explore/components/DatasourcePanel.tsx (hunk 2 above)
Comment (pkdotson) [reply superset-frontend/src/explore/components/DatasourcePanel.tsx:134]: yep!

**Pull request merged**